federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -28,11 +28,11 @@ import (
 
 	federationclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset"
 	"k8s.io/kubernetes/federation/cmd/federation-controller-manager/app/options"
-	"k8s.io/kubernetes/federation/pkg/dnsprovider"
 	clustercontroller "k8s.io/kubernetes/federation/pkg/federation-controller/cluster"
 	configmapcontroller "k8s.io/kubernetes/federation/pkg/federation-controller/configmap"
 	daemonset "k8s.io/kubernetes/federation/pkg/federation-controller/daemonset"
 	deploymentcontroller "k8s.io/kubernetes/federation/pkg/federation-controller/deployment"
+	dnscontroller "k8s.io/kubernetes/federation/pkg/federation-controller/dns"
```

Comment:
I don't have strong inclination towards both the names, "dns-controller" or "service-dns-controller". just that in future maybe we need to maintain dns entries for federated-ingresses. just not clear for me as of now. I need your final say and would go with that name :)